core/src/actors/chain_manager/mod.rs

```diff
@@ -83,6 +83,8 @@ pub struct ChainManager {
     epoch_to_block_hash: HashMap<Epoch, HashSet<Hash>>,
     /// Map that stores blocks by their hash
     blocks: HashMap<Hash, Block>,
+    /// Current Epoch
+    current_epoch: Epoch,
```

Comment:
This should probably be an `Option<Epoch>` (to handle the edge case when we don't know the current epoch), and it should be updated by the `Handler<EpochNotification<EveryEpochPayload>>`, like so:

```rust
fn handle... {
    self.current_epoch = Some(msg.checkpoint);
```